src/freenet/node/useralerts/UserAlertManager.java

```diff
@@ -243,11 +244,37 @@ public HTMLNode renderAlert(UserAlert userAlert) {
 		userAlertNode.addChild("div", "class", "infobox-header", userAlert.getTitle());
 		HTMLNode alertContentNode = userAlertNode.addChild("div", "class", "infobox-content");
 		alertContentNode.addChild(userAlert.getHTMLText());
+		if (userAlert instanceof UserAlertFromPeer) {
+			UserAlertFromPeer n2nUserAlert = (UserAlertFromPeer) userAlert;
+			alertContentNode.addChild(renderReplyButton(n2nUserAlert, n2nUserAlert.getSourceNode()));
```

Comment:
If you are already handing in the `UserAlertFromPeer`, why are you extracting the source node and handing that in separately?